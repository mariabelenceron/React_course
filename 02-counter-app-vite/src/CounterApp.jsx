import { useState } from 'react'  //Hook

import PropTypes from 'prop-types'



// Functional Component
export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState(value)
    
    const handleAdd = () => {
        // setCounter(counter + 1)
        setCounter((c) => c + 1)
    }
    
    const handleSubtract = () => setCounter(counter - 1)
    
    const handleReset = () => setCounter(value)

    return ( 
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ handleSubtract }>-1</button>
            <button aria-label='btn-reset' onClick={ handleReset }>Reset</button>
            <button onClick={ handleAdd }> +1 </button>
        </>
    );
}


// Configuraciones
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}