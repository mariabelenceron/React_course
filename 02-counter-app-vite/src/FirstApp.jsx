import PropTypes from 'prop-types';

//--------- Variables
const newMessage = {
    name: 'Belen',
    note: 'Es mi primera app en React'
};

//--------- Functions: Ponerlas afuera para evitar que el FC la renderice de nuevo y utilice mas espacio de memoria
const greeting = ({name, note}) => {
    return `Hola ${name}, tu nota es: ${note}!`
}

// Functional Component
// props : Un objeto, permite establecer una comunicacion entre el componente padre hacia el componente hijo
export const FirstApp = ( {title, subtitle} ) => {
    // console.log(props);

    return (
        // Fragmento
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{ greeting(newMessage) }</p>
        </>
    )
}


// Configuraciones al componente
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 'No hay subtitulo'
}