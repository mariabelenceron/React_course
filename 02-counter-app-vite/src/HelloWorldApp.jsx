// Functional Component
export const HelloWorldApp = () => {
    return (
        <div>HelloWorldApp</div>
    )
}
