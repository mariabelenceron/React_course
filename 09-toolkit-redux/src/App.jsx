import { useDispatch, useSelector } from 'react-redux'
import './App.css'
import { decrement, increment, incrementByAmount } from './store/slices/counter'

function App() {

  const { counter } = useSelector(state => state.counter)
  const dispatch = useDispatch()

  return (
    <>
      <h1>Vite + React</h1>
      <div className="card">
        <p>count is {counter}</p>
        <button onClick={() => dispatch(increment())}>
          Increment
        </button>
        <button onClick={() => dispatch(decrement())}>
          Decrement
        </button>
        <button onClick={() => dispatch(incrementByAmount(2))}>
          Increment by 2
        </button>
      </div>
    </>
  )
}

export default App
