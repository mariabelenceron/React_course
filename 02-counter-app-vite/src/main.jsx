import React from 'react';

import ReactDOM from 'react-dom/client'; // Se usa para renderizar la aplicacion

import {CounterApp} from './CounterApp';
import { FirstApp } from './FirstApp';

// import { HelloWorldApp } from './HelloWorldApp';

import './styles.css'; //Tener estilos en toda la aplicaci√≥n


// Renderizar el documento
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* Properties (title) 
            boolean = true de manera implicita
            */}

        {/* <FirstApp title="Hola" subtitle="Soy Belen!"/> */}

        <CounterApp value={1}/>
    </React.StrictMode>
)
