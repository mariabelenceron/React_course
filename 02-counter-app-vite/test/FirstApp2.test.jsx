import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    const title = "Hola"
    const subtitle = "Soy Belen!"

    test('debe hacer match con el snapshot', ()=>{
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    })
    
    test('debe mostrar el mensaje "Soy Belen!"', () => {
        render(<FirstApp title={title}/>);
        // screen.debug()
        expect(screen.getByText(title)).toBeTruthy();
    });
    
    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title)
    });

    test('debe mostrar el subtitulo enviado por props', () => {
        render(
            <FirstApp 
                title={title}
                subtitle={subtitle}
            />
        );
        expect(screen.getAllByText(subtitle).length).toBe(2);

    });
});