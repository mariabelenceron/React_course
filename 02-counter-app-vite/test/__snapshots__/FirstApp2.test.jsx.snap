// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Hola Belen, tu nota es: Es mi primera app en React!
  </p>
</div>
`;
